Fix warnings when building gradle plugin

`$ ./gradlew clean goJF build` causes two types of warnings at the moment.

- [x] Unchecked future result
```
> Task :compileTestJava 
(...)/jib/jib-core/src/test/java/com/google/cloud/tools/jib/http/TestWebServer.java:40: warning: [FutureReturnValueIgnored] Return value of methods returning Future must be checked. Ignoring returned Futures suppresses exceptions thrown from the code that completes the Future.
    executorService.submit(this::serve200);
                          ^
    (see http://errorprone.info/bugpattern/FutureReturnValueIgnored)
  Did you mean to remove this line?
1 warning
```

- [x] `@Input` annotation used on `java.io.File`
```
> Task :validateTaskProperties 
Task property validation finished with warnings:
  - Warning: Task type 'com.google.cloud.tools.jib.gradle.BuildDockerTask': property 'jib.extraDirectory' has @Input annotation used on property of type java.io.File.
  - Warning: Task type 'com.google.cloud.tools.jib.gradle.BuildImageTask': property 'jib.extraDirectory' has @Input annotation used on property of type java.io.File.
  - Warning: Task type 'com.google.cloud.tools.jib.gradle.BuildTarTask': property 'jib.extraDirectory' has @Input annotation used on property of type java.io.File.
  - Warning: Task type 'com.google.cloud.tools.jib.gradle.DockerContextTask': property 'jib.extraDirectory' has @Input annotation used on property of type java.io.File.
```

--- a/jib-core/src/test/java/com/google/cloud/tools/jib/http/TestWebServer.java
+++ b/jib-core/src/test/java/com/google/cloud/tools/jib/http/TestWebServer.java
@@ -24,6 +24,7 @@ import java.net.Socket;
 import java.nio.charset.StandardCharsets;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
 import java.util.concurrent.Semaphore;
 
 /** Simple local web server for testing. */
@@ -35,7 +36,7 @@ class TestWebServer implements Closeable {
 
   TestWebServer() throws IOException, InterruptedException {
     serverSocket = new ServerSocket(0);
-    executorService.submit(this::serve200);
+    ignoreReturn(executorService.submit(this::serve200));
     threadStarted.acquire();
   }
 
@@ -61,5 +62,9 @@ class TestWebServer implements Closeable {
       for (int ch = in.read(); ch != -1; ch = in.read()) ;
     }
     return null;
+  }
+
+  private void ignoreReturn(Future<Void> future) {
+    // do nothing; to make Error Prone happy
   }
 }

